gitgud/operations.py

```diff
@@ -18,19 +18,16 @@
 from gitgud.hooks import all_hooks
 
 
-class DirectoryContent:
-    def __init__(self, content):
-        self.content = content
-
+class DirectoryContent(dict):
```

Comment:
I like that we're now subclassing. What additional features do we get from this?

Reply:
The reason why I wanted to do this is because I used the `keys()` method from when `get_*_content` returned a Dictionary for the `content_display_data` method. We can also use anything else that a Dict object could use.

Reply:
Are there any potential issues with this? Would we possibly want to override assigning so that this class automatically can convert paths to keys `dir_content[Path('stuff')] = "Hello world"` can be looked up such that `dir_content['stuff'] == "Hello world"`

Reply:
Hmm, that would be an issue in an abstract sense, but I don't see a benefit of assigning to DirectoryContent. Since the object only serves as a window into the content of the working directory/staging area/commits, DirectoryContent seems only useful in a read-only sense.

That being said, since we're not *assigning* to DirectoryContent values yet (since we're only using it to check pre-existing file content right now), implementing a `__setitem__` that converts Paths to str would be relatively harmless at this point. So, up to you if you think it's worth implementing

Reply:
I think it's unnecessary. If it comes up later, we can add it.